kamon-akka-sharding/src/main/scala/kamon/akka/sharding/metrics/AkkaShardMetrics.scala

```diff
@@ -0,0 +1,24 @@
+package kamon.akka.sharding.metrics
+
+import akka.cluster.sharding.ShardRegion.ShardState
+import kamon.metric.instrument.InstrumentFactory
+import kamon.metric.{ EntityRecorderFactory, GenericEntityRecorder }
+
+class AkkaShardMetrics(instrumentFactory: InstrumentFactory) extends GenericEntityRecorder(instrumentFactory) {
+
+  val numShardsGauge = gauge("number-of-shards", 0L)
+
+  def record(shardStats: Set[ShardState]) = {
+    numShardsGauge.record(shardStats.size)
+    shardStats.foreach(state ⇒ {
+      val shardGauge = gauge(s"number-of-entities-shard-${state.shardId}", 0L)
+      shardGauge.record(state.entityIds.size)
```

Comment:
After some more testing, this doesn't seem to work. It does not update the current value of the `CurrentValueCollector`. So when the value is refreshed by the internal gauge refresh schedule, it will record the initial value for all subsequent invocations. I guess I either need to create a `Gauge` that is never refreshed (only recorded), or create a custom `CurrentValueCollector` that can be updated from within this method.